lib/auth/github.go

```diff
@@ -952,7 +956,26 @@ func ValidateClientRedirect(clientRedirect string, ssoTestFlow bool, settings *t
 	case "http":
 		switch u.Hostname() {
 		default:
-			return trace.BadParameter("invalid hostname in client redirect URL")
+			if settings == nil || len(settings.InsecureAllowedCidrRanges) == 0 {
+				return trace.BadParameter("invalid hostname in client redirect URL")
+			}
+			hostIP, err := netip.ParseAddr(u.Hostname())
+			if err != nil {
+				return trace.BadParameter("custom client HTTP redirect URLs must be IP addresses")
+			}
+
```

Comment:
We should probably treat IPv6-mapped IPv4 addresses as IPv4.

```suggestion
			}
			hostIP = hostIP.Unmap()

```